packages/cli/tests/e2e/bot/Arm_Bot_CanDeployToCustomizedRg.tests.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author Di Lin <dilin@microsoft.com>
+ */
+
+import path from "path";
+
+import { AadValidator, BotValidator } from "../../commonlib";
+import {
+  getSubscriptionId,
+  getTestFolder,
+  getUniqueAppName,
+  cleanUp,
+  readContextMultiEnv,
+  createResourceGroup,
+  deleteResourceGroupByName,
+  setBotSkuNameToB1Bicep,
+} from "../commonUtils";
+import AppStudioLogin from "../../../src/commonlib/appStudioLogin";
+import { environmentManager, isFeatureFlagEnabled } from "@microsoft/teamsfx-core";
+import { FeatureFlagName } from "@microsoft/teamsfx-core/src/common/constants";
+import { CliHelper } from "../../commonlib/cliHelper";
+import { Capability, ResourceToDeploy } from "../../commonlib/constants";
+import { customizeBicepFilesToCustomizedRg } from "../commonUtils";
+
+describe("Deploy to customized resource group", function () {
+  //  Only test when insider feature flag enabled
+  if (!isFeatureFlagEnabled(FeatureFlagName.InsiderPreview, true)) {
+    return;
+  }
+
+  const testFolder = getTestFolder();
+  const subscription = getSubscriptionId();
+  let appName: string, projectPath: string;
+
+  beforeEach(async () => {
+    appName = getUniqueAppName();
+    projectPath = path.resolve(testFolder, appName);
+  });
+
+  afterEach(async () => {
+    await cleanUp(appName, projectPath, true, false, false, true);
+  });
+
+  it(`bot project can deploy bot resource to customized resource group and successfully provision / deploy`, async function () {
+    // Create new bot project
+    await CliHelper.createProjectWithCapability(appName, testFolder, Capability.Bot);
+
+    // Create empty resource group
+    const customizedRgName = `${appName}-customized-rg`;
+    await createResourceGroup(customizedRgName, "eastus");
+
+    // Customize simple auth bicep files
+    await customizeBicepFilesToCustomizedRg(
+      customizedRgName,
+      projectPath,
+      `name: 'botProvision'`,
+      `name: 'addTeamsFxBotConfiguration'`
+    );
+
+    // Provision
+    setBotSkuNameToB1Bicep(projectPath, environmentManager.getDefaultEnvName());
+    await CliHelper.setSubscription(subscription, projectPath);
+    await CliHelper.provisionProject(projectPath);
+
+    // deploy
+    await CliHelper.deployProject(ResourceToDeploy.FrontendHosting, projectPath);
```

Comment:
Ask a question to clarify my confusion - In this case, provision bot but deploy frontend?

Reply:
fixed. Here we need to deploy bot. And this cli [bug](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12905612) hides this bug without throwing out error. @Alive-Fish will help track the bug.